### ISSUE TITLE
Go package version 1.18 not being installed with helm 3.9.1.

### ISSUE THREAD

**Issue by Sagargowdatp** [helm/helm#11155]
Even after the successful installation of helm 3.9.1 Go version is not yet updated and it still remains with version **1.17.5.**

**version.BuildInfo{Version:"v3.9.1", GitCommit:"a7c043acb5ff905c261cfdc923a35776ba5e66e4", GitTreeState:"clean", GoVersion:"go1.17.5"}**

Expectation: Go version **1.18** need to be included with helm 3.9.1. as per #11110

Issue :go version 1.17.5  has several high severity [CVEs:]
CVE-2022-23772
CVE-2022-23806
CVE-2022-23773
CVE-2022-24921
CVE-2022-24675
CVE-2022-28327

please help in resolving the issue.

**Comment by bacongobbler** [comment]
The only CVE in that list that is relevant to Helm users is [CVE-2022-24921](https://github.com/advisories/GHSA-6685-ffxp-xm6f) for `helm search --regexp` and `helm list --filter`, as those are the only locations where we accept user input for `regexp.Compile`. Neither of those are of any cause for concern because that exploit forces the program to exit, and it requires the user's input. In other words, if you copy/paste a regular expression into `helm search --regexp` that contains a deeply nested expression, the program will exit. No harm is caused to the system.

In other words, these CVEs are a non-issue and do not require an urgent security fix.

Helm 3.10.0 is scheduled for September 14th. That release will contain the update to Go 1.18.
